src/plugins/Polls/UI/Polls.tsx

```diff
@@ -0,0 +1,128 @@
+import React from 'react'
+import { makeStyles, createStyles, Card, Typography, CircularProgress } from '@material-ui/core'
+import type { PollGunDB } from '../Services'
+import { format, isValid } from 'date-fns'
+
+const useStyles = makeStyles((theme) =>
+    createStyles({
+        card: {
+            borderRadius: theme.spacing(1),
+            margin: theme.spacing(2, 0),
+            padding: theme.spacing(2),
+        },
+        line: {
+            display: 'flex',
+            justifyContent: 'space-between',
+        },
+        status: {
+            display: 'flex',
+            alignItems: 'center',
+        },
+        statusText: {
+            margin: '3px',
+            fontSize: '13px',
+            color: theme.palette.primary.main,
+        },
+        option: {
+            position: 'relative',
+            display: 'flex',
+            justifyContent: 'space-between',
+            margin: theme.spacing(1, 0),
+            padding: theme.spacing(0, 1),
+            height: '28px',
+        },
+        bar: {
+            position: 'absolute',
+            top: '0',
+            left: '0',
+            zIndex: 100,
+            backgroundColor: theme.palette.primary.main,
+            opacity: 0.6,
+            minWidth: theme.spacing(1),
+            height: '28px',
+            borderRadius: theme.spacing(0.8),
+        },
+        text: {
+            zIndex: 101,
+            lineHeight: '28px',
+            margin: '0 4px',
+        },
+        deadline: {
+            color: '#657786',
+        },
+    }),
+)
+
+interface PollCardProps {
+    poll: PollGunDB
+    onClick?(): void
+    vote?(poll: PollGunDB, index: number): void
+    status?: PollStatus
+}
+
+export enum PollStatus {
+    Voted = 'Voted.',
+    Voting = 'Voting',
+    Error = 'Error',
+    Closed = 'Closed',
+    Inactive = 'Inactive',
+}
+
+export function PollCardUI(props: PollCardProps) {
+    const { poll, onClick, vote, status } = props
+    const classes = useStyles()
+    const isClosed = Date.now() > poll.end_time ? true : false
+
+    const totalVotes = poll.results.reduce(
+        (accumulator: number, currentValue: number): number => accumulator + currentValue,
+    )
+
+    const getDeadline = (date: number) => {
+        const deadline = new Date(date)
+        if (isValid(deadline)) {
+            return `Deadline: ${format(deadline, 'yyyy-MM-dd HH:mm:ss')}`
```

Comment:
"${time} left until vote ends." You may want to use https://mdn.io/Intl.RelativeTimeFormat to calculate that

Reply:
good idea

Reply:
I want to handle it like this: When there are less than 24 hours left, the text is shown as "${time} left until vote ends.".Ohterwise, it's shown as `Deadline: ${format(deadline, 'yyyy-MM-dd HH:mm')}`

Reply:
![](https://user-images.githubusercontent.com/5390719/93602481-3e8db700-f9f5-11ea-9645-6aa88ca42149.png)

actually you don't need to do anything.

Reply:
okk

Reply:
![image](https://user-images.githubusercontent.com/48639990/93607920-8ebc4780-f9fc-11ea-9c68-8ef131258e7b.png)
It seems that typescript don't support this property...

Reply:
Maybe I can use https://date-fns.org/v2.16.1/docs/formatDistance

Reply:
Please use `as any` as a workaround and paste this link: microsoft/TSJS-lib-generator#917

Reply:
With `Intl.RelativeTimeFormat.format()` I need to calculate the number of days\hours\minutes needed, why not use the `formatDistance` of date-fns directly
```typescript
            const lang = useValueRef(languageSettings)
            const localeMapping = () => {
                switch (lang) {
                    case 'en':
                        return enUS
                    case 'ja':
                        return ja
                    case 'zh':
                        return zhTW
                    default:
                        return enUS
                }
            }
            const time = formatDistance(new Date(poll.start_time), new Date(poll.end_time), {
                locale: localeMapping(),
            })
            return t('plugin_poll_deadline', { time })
```